src/state.cairo

```diff
@@ -12,62 +14,27 @@ pub struct ChainState {
     /// Total work done.
     pub total_work: u256,
     /// Best block.
-    pub best_block_hash: u256,
+    pub best_block_hash: Hash,
     /// Current target.
     pub current_target: u256,
     /// Start of the current epoch.
     pub epoch_start_time: u32,
     /// Previous timestamps.
     pub prev_timestamps: Span<u32>,
-    /// Utreexo state
+    /// Utreexo state.
     pub utreexo_state: UtreexoState,
 }
 
-/// Accumulator representation of the state aka "Compact State Node"
-#[derive(Drop, Copy)]
-pub struct UtreexoState {
-    /// Roots of Merkle tree forest
-    pub roots: Span<felt252>,
-}
-
-/// Utreexo set is used to retrieve TXOs spent by particular inputs
-#[derive(Drop, Copy)]
-pub struct UtreexoSet {
-    /// A list of extended transaction outputs spent in a particular block(s).
-    pub outputs: Span<UtreexoOutput>,
-}
-
-/// TXO extended with info about parent transaction and the position within it.
-/// The hash of this structure is a leaf node in the Utreexo Merkle tree forest.
-#[derive(Drop, Copy)]
-pub struct UtreexoOutput {
-    /// The TXID this output belongs to.
-    pub txid: u256,
-    /// The index of this output.
-    pub vout: u32,
-    /// Output data
-    pub output: TxOut,
-}
-
-/// Inclusion proof for multiple leaves
-#[derive(Drop, Copy)]
-pub struct UtreexoBatchProof {
-    /// Indices of tree leaves, one for each output in the utreexo set
-    pub targets: Span<u64>,
-    /// All the nodes required to calculate the root
-    pub proof: Span<felt252>,
-}
-
 /// Represents a block in the blockchain.
```

Comment:
here i'm just modifying order of structs in state.cairo to make it more logical

State -> block -> header -> Transaction -> TxIn -> OutPoint -> TxOut